### PR TITLE
updated some of the phrasing around hello-world per issue number 14960

### LINKED ISSUE #14960: Improve the Hello World guide
### What article on docs.github.com is affected?

https://docs.github.com/en/get-started/quickstart/hello-world

### What part(s) of the article would you like to see updated?

We received the following comments about the Hello World guide. Overall the guide gets good feedback, but it's important to keep in mind that for many users, this is their first introduction to GitHub and so we don't want to confuse or intimidate them with jardon and instructions without context.

> the merge pull request section did not make sense.

Add some additional screenshots to give this section more context.

> Hard to understand.
> Make the text simplified for people who knows nothing about coding.
> keep your descriptions/ guides, easy to understand by avoiding high-level vocabulary

We should [link to the glossary](https://docs.github.com/en/get-started/quickstart/github-glossary) the first time any of these terms are used.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by cmwilson21**
### Why:
This is an attempt at updating a bit of the Hello World walk-through guide
Closes [issue link](https://github.com/github/docs/issues/14960)

### What's being changed:
I added a little more explanations to certain areas and tried to specifically address the merge pull request section as requested by issue number 14960.

### Check off the following:

- [ x] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [ x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by cmwilson21** [comment]
I went step by step through the Hello World guide and followed it, adding and adjusting to the doc as I went along.

**Comment by ramyaparimi** [comment]
@cmwilson21 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: content/get-started/quickstart/hello-world.md [hunk 1]
```diff
@@ -116,18 +117,18 @@ You can even open pull requests in your own repository and merge them yourself.
...
 7. Click **Create pull request**.
 
 Your collaborators can now review your edits and make suggestions.
 
 ## Merging your pull request
 
-In this final step, you will merge your `readme-edits` branch into the `main` branch.
+In this final step, you will merge your `readme-edits` branch into the `main` branch. After you created the pull request in the above step, you will be brought to a page that shows you any conflicts your code may have with the branch you are merging to - the `main` branch. You can walk through those changes and conflicts and address them with your team members in the comments. For our purposes in this walk-through, there are no conflicts, so you can go ahead and merge your branch into the main branch. You’ll have one more chance to change or edit your commit message.
```
Comment (ethanpalm) [inline content/get-started/quickstart/hello-world.md:127]:
```suggestion
In this final step, you will merge your `readme-edits` branch into the `main` branch.  After you merge your pull request, the changes on your `readme-edits` branch will be incorporated into `main`.

Sometimes, a pull request may introduce changes to code that conflict with the existing code on `main`. If there are any conflicts, {% data variables.product.product_name %} will alert you about the conflicting code and prevent merging until the conflicts are resolved. You can make a commit that resolves the conflicts or use comments in the pull request to discuss the conflicts with your team members.

In this walk-through, you should not have any conflicts, so you are ready to merge your branch into the main branch.
```

I think we should separate the information about merge conflicts. It's important info, but might be a lot for someone new to understand at once. The main thing I think we need to emphasize here is that merging is what brings your changes from a branch into `main`.

File: content/get-started/quickstart/hello-world.md [hunk 2]
```diff
@@ -116,18 +117,18 @@ You can even open pull requests in your own repository and merge them yourself.
...
 
 1. Click **Merge pull request** to merge the changes into `main`.
-2. Click **Confirm merge**.
-3. Go ahead and delete the branch, since its changes have been incorporated, by clicking **Delete branch**.
+2. Click **Confirm merge**. You'll receive a message that the request was successfully merged and the request was closed.
```
Comment (ethanpalm) [inline content/get-started/quickstart/hello-world.md:130]:
```suggestion
2. Click **Confirm merge**. You will receive a message that the request was successfully merged and the request was closed.
```

File: content/get-started/quickstart/hello-world.md [hunk 3]
```diff
@@ -116,18 +117,18 @@ You can even open pull requests in your own repository and merge them yourself.
...
+3. You won't have any need for the `readme-edits` branch anymore. So, go ahead and delete the branch, since its changes have been incorporated, by clicking **Delete branch**. You can always create a new branch and repeat the process when needed.
```
Comment (ethanpalm) [inline content/get-started/quickstart/hello-world.md:131]:
```suggestion
3. Click **Delete branch**. Now that your pull request is merged and your changes are on `main`, you can safely delete the `readme-edits` branch. If you want to make more changes to your project, you can always create a new branch and repeat this process.
```

**Review by ethanpalm — CHANGES_REQUESTED** [review]
Thanks for creating this PR, @cmwilson21 🎉 

These changes are looking great! I think we just need a few additional items and this will be ready to merge.

1. In the "Introduction" section, it would be great to include a link to https://docs.github.com/en/get-started/quickstart/github-glossary so people can reference the new terminology that might be unfamiliar to them. I think it would fit in the last paragraph well, but totally up to you ⚡ 
2. I think a screenshot will really help people in the "Merging your pull request" section. Kind of like how "[Creating a repository](https://docs.github.com/en/get-started/quickstart/hello-world#creating-a-repository)" has an image at the end that shows all of the various options that are mentioned in the procedure. Are you able to add screenshots to articles? I'm not entirely sure what permissions you have to the repo 😅 

Please let me know if any of my comments on the existing changes aren't clear or you have any questions ✍🏼

**Comment by cmwilson21** [comment]
@ethanpalm Thanks for the suggestions and advice! Everything you've mentioned makes sense and I've taken notes for future reference. I appreciate you taking the time to explain things to me. I'll get to work on committing these changes!

**Comment by cmwilson21** [comment]
Added the suggested reference to the glossary in the introduction section.

**Comment by cmwilson21** [comment]
Hey @ethanpalm, I'm not 100% certain on the screenshot issue. I'm looking into it.

File: content/get-started/quickstart/hello-world.md [hunk 4]
```diff
@@ -117,17 +118,22 @@ You can even open pull requests in your own repository and merge them yourself.
...
 ## Merging your pull request
 
-In this final step, you will merge your `readme-edits` branch into the `main` branch.
+In this final step, you will merge your `readme-edits` branch into the `main` branch.  After you merge your pull request, the changes on your `readme-edits` branch will be incorporated into `main`.
+
+Sometimes, a pull request may introduce changes to code that conflict with the existing code on `main`. If there are any conflicts, {% data variables.product.product_name %} will alert you about the conflicting code and prevent merging until the conflicts are resolved. You can make a commit that resolves the conflicts or use comments in the pull request to discuss the conflicts with your team members.
+
+In this walk-through, you should not have any conflicts, so you are ready to merge your branch into the main branch.
 
 1. Click **Merge pull request** to merge the changes into `main`.
```
Comment (cmwilson21) [inline content/get-started/quickstart/hello-world.md:134]:
```suggestion
1. Click **Merge pull request** to merge the changes into `main`.
  ![Screen shot of merge button.](/assets/images/help/pull_requests/pullrequest-mergebutton.png)
```

**Review by ethanpalm — APPROVED** [review]
These changes look great. I'll get this merged once all checks are passing 🎉

**Pull request merged**